src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1068,6 +1080,68 @@
     "alter table reports add column corrective_change boolean"
     "alter table resource_events add column corrective_change boolean"))
 
+(defn resource-params-cache-parameters-to-jsonb
```

Comment:
They're both relatively quick with the 5000 node set we have, which has 136926 fact values but only 711 cached resource params. I don't think that's representative enough. I'll try and get some more user info today to see what's appropriate.

I think the memory issue can be reduced by streaming the select through the application into a new table before deleting the old one, but I haven't figured out a way to avoid the trip through the application since postgres doesn't seem to have the function I need to go from a json-serialized string to proper json. Will keep looking for that as well.